src/game_battlealgorithm.cpp

```diff
@@ -766,6 +773,7 @@ void Game_BattleAlgorithm::AlgorithmBase::TargetFirst() {
 	} else {
 		if (IsReflected()) {
 			original_targets.push_back(GetTarget());
+			current_original_target = original_targets.begin();
```

Comment:
Do we have to store this iterator? Can't you change `GetOriginalTarget()` to `GetFirstOriginalTarget()` and return the first one, or change it to `GetOriginalTargets()` and return a reference to the whole vector and let the caller decide what to do with it.